fix(rn-tf): Updated 2.28 RN and Compat Matrix for correct TF version support

tsonny opened the pull request:
Resolves Jira: [https://armory.atlassian.net/browse/BOB-30827]

tsonny committed:
marketing blurb back - and some other fixes and edits

tsonny committed:
docs(tf-version): updated rn + compat-matrix

mknous reviewed (commented):
It looks as if we have to use something other than Armory CD in the compat matrix - I am not sure what we are consistently using for the product names in our external facing documentation.

tsonny committed:
removed unwanted changes from PR "Updated 1 path from 513e426f
tsonny@tiffanysonny-mbp docs % git status
On branch BOB-30827
Your branch is up to date with 'origin/BOB-30827'."

tsonny commented:
Unintended commit included - closing and will push new PR with the correct changes only.
